lib/redirect.js

```diff
@@ -0,0 +1,34 @@
+import { useEffect } from "react"
+import { useRouter } from "next/router"
+import languageDetector from "./languageDetector"
+
+export const useRedirect = (to) => {
+  const router = useRouter()
+  to = to || router.asPath
+
+  // language detection
+  useEffect(() => {
+    const detectedLng = languageDetector.detect()
+    if (to.startsWith(`/${detectedLng}`) && router.route === "/404") {
+      // prevent endless loop
+      router.replace(`/${detectedLng}${router.route}`)
+      return
+    }
+
+    languageDetector.cache(detectedLng)
+    router.replace(`/${detectedLng}${to}`)
+  })
+
+  return <></>
```

Comment:
maybe this: return <></> is required by the library, but can you explain me why this return is needed?

Reply:
Of course, useRedirect, Redirect and getRedirect are components, and in all React components you always have to return some jsx, otherwise you return jsx, for that these returns. In React if we need return component empty, we can use "<></>", it is shorthand syntax for that. I hope you that it helps you this explain